### ISSUE TITLE
Crash when editing static encounters

### ISSUE THREAD

**Issue by Geodude6** [kwsch/pk3DS#382]
I'm running into an issue with the static encounter editor. When I change the Pokemon used for the encounter, I get this error message (pasted below). I suspect this is related to me hex editing a/1/5/9, as that's when I first started running into this error. I got this specific error when I tried to change the Island Scan Rotom in Ultra Moon to Arceus:

```
Exception Details:
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at pk3DS.StaticEncounterEditor7.GetAllies() in C:\BuildAgent\work\ccf72a59f9911202\pk3DS\Subforms\Gen7\StaticEncounterEditor7.cs:line 253
   at System.Windows.Forms.ListBox.OnSelectedIndexChanged(EventArgs e)
   at System.Windows.Forms.ListBox.NativeRemoveAt(Int32 index)
   at System.Windows.Forms.ListBox.ObjectCollection.SetItemInternal(Int32 index, Object value)
   at System.Windows.Forms.ComboBox.OnSelectedIndexChanged(EventArgs e)
   at System.Windows.Forms.ComboBox.WmReflectCommand(Message& m)
   at System.Windows.Forms.ComboBox.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

**Comment by kwsch** [comment]
Have you tried restoring to an umodified file?

**Comment by Geodude6** [comment]
Just tried it and the issue persists.

Exception Details:
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at pk3DS.StaticEncounterEditor7.GetAllies() in C:\BuildAgent\work\ccf72a59f9911202\pk3DS\Subforms\Gen7\StaticEncounterEditor7.cs:line 253
   at System.Windows.Forms.ListBox.OnSelectedIndexChanged(EventArgs e)
   at System.Windows.Forms.ListBox.NativeRemoveAt(Int32 index)
   at System.Windows.Forms.ListBox.ObjectCollection.SetItemInternal(Int32 index, Object value)
   at System.Windows.Forms.ComboBox.OnSelectedIndexChanged(EventArgs e)
   at System.Windows.Forms.ComboBox.WmReflectCommand(Message& m)
   at System.Windows.Forms.ComboBox.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

Loaded Assemblies:
--------------------
mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
C:\Windows\Microsoft.NET\Framework64\v4.0.30319\mscorlib.dll

pk3DS, Version=1.0.0.379, Culture=neutral, PublicKeyToken=null
C:\Users\mhollmann\Desktop\New folder\Pokemon\pk3ds\pk3DS.exe

System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System\v4.0_4.0.0.0__b77a5c561934e089\System.dll

System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Windows.Forms\v4.0_4.0.0.0__b77a5c561934e089\System.Windows.Forms.dll

System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Drawing\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Drawing.dll

System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Core\v4.0_4.0.0.0__b77a5c561934e089\System.Core.dll

System.Configuration, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Configuration\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Configuration.dll

System.Xml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Xml\v4.0_4.0.0.0__b77a5c561934e089\System.Xml.dll

pk3DS.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
C:\Users\mhollmann\Desktop\New folder\Pokemon\pk3ds\pk3DS.Core.dll

--------------------
User Message:
An unhandled exception has occurred.
You can continue running the program (albeit with potential side-effects), but please report this error.

**Comment by Geodude6** [comment]
I just deleted and redownloaded pk3ds and I still run into the same problem regardless of whether I try it on the modified file or the unmodified one.

**Comment by kwsch** [comment]
Judging from the [file's history](https://github.com/kwsch/pk3DS/commits/master/pk3DS/Subforms/Gen7/StaticEncounterEditor7.cs), paging @sora10pls 

Probably just need to check the (entry < 0) and early return

**Comment by kwsch** [comment]
Resolved by #386
